src/datasources/data-source.ts

```diff
@@ -0,0 +1,124 @@
+import { default as booleanDisjoint } from '@turf/boolean-disjoint';
+import { polygon } from '@turf/helpers';
+import * as _ from 'lodash';
+
+export class DataSource {
+  // Type could be local, in the future could be 'https'
+  type: string;
+  // Input. Just for type local
+  localInput: any;
+
+  // opts = {
+  //  input: { stream: null, format: 'geojson' } ,
+  //
+  // }
+
+  constructor(type: string, opts: any) {
+    this.type = type;
+    if (type === 'local') {
+      this.localInput = opts.input;
+    }
+  }
+
+  // This method handles the data (either coming from this.data.stream or from a service)
+  // type: could be 'histogram' or 'variable'
+  // opts: { bbox: [turf polygon], agg: 'count(*)', property: 'age'}
+  fetch(type: string, opts: any) {
+    return new Promise((resolve, reject) => {
+      if (this.type === 'local') {
+        const data = this.applyFilters(opts);
+        const agg = this.parseAgg(opts.agg);
+        if (type === 'histogram') {
+          resolve(this.groupBy(data, opts.property, agg));
+        } else if (type === 'variable') {
+          resolve( {
+            value: this.aggregator(data, agg)
+          });
+        } else if (type === 'stacked') {
+          resolve(rankingDoubleHistogram(data));
```

Comment:
Este código debería crear un histograma a partir de los datos crudos. Ahora lo que hace es coger un dobleHistogram y cambiarle el formato. No es la idea. 

Haciendo esto lo que hacemos es meter la lógica de reveal en latitude.

Si queréis por ahora dejarlo en blanco.

Reply:
Dejamos solo el resolve y ta vemos como hacer las funciones de agrupación